pkg/utils/history_buffer.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import "fmt"
+
+type HistoryBuffer[T any] struct {
+	maxSize int
+	Items   []T
+}
+
+func NewHistoryBuffer[T any](maxSize int) *HistoryBuffer[T] {
+	return &HistoryBuffer[T]{
+		maxSize: maxSize,
+		Items:   make([]T, 0, maxSize),
+	}
+}
+
+func (self *HistoryBuffer[T]) Push(item T) {
+	if len(self.Items) == self.maxSize {
+		self.Items = self.Items[:len(self.Items)-1]
+	}
+	self.Items = append([]T{item}, self.Items...)
+}
+
+func (self *HistoryBuffer[T]) Pop() (T, error) {
```

Comment:
This is dead code (and I don't imagine we would ever need to pop from the stack) so I'd remove it

Reply:
true, true

removed it in: 8e73ffe7d5f43b0b51a293c0c3096f93d5966392